File: all-submissions/1046-max-consecutive-ones-iii/solution.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int countZeros = 0;
        int left =0;
        int right =0;
        int maxOnes = 0;
        while(right < nums.length) {
            if(nums[right] == 0) {
                countZeros++;
            }
            if(countZeros > k) {
                while(countZeros > k && left < nums.length && left <= right){
                    if(nums[left] == 0) {
                        countZeros--;
                    } 
                    left++;
                }
            }
            maxOnes = Math.max(maxOnes, right-left+1);
            right++;
        }
        return maxOnes;
    }
}

File: all-submissions/1046-max-consecutive-ones-iii/solution.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        left = 0
        right = 0
        countZero = 0
        max_ones =0
        while right < len(nums):
            if nums[right] == 0:
                countZero +=1
                while countZero > k:
                    if nums[left] == 0:
                        countZero -= 1
                    left+=1
            max_ones = max(max_ones, right-left+1)
            right +=1
        return max_ones
